### DISCUSSION TITLE
Adaptions for 3.3V?

### DISCUSSION THREAD

**Discussion by jangerd** [AK-Homberger/Seatalk-Autopilot-Remote-Control#11] [Ideas]
First of all, thank you very much for your work, it is fun and really useful project.

What I would like to do, however, is use it with an ESP32 instead of the Ardiuno which opens the door to steer your boat via mobile (bluetooth). Being a software engineer with little understanding of voltage deviders, could you think of a solution (probably just replacement of resistors!?) that works with 3.3V?

Thank you and viele Grüße
Jan-Gerd

**Comment by AK-Homberger** [comment]
Hi Jan-Gerd,

the Voltage devider can be changed to ensure the high level is below 3.3 Volt (which is usually the case anway). 

The other problem is the availabilty of a 9-Biit serial protocol for the ESP32. I'm not aware of any library with this function. But that does not mean it does not exist. I simply never looked for that function so far.

Regards,
Andreas

**Reply by jangerd** [reply]
Hi Andreas,

Thank you for the instant reply. The ESP32 IDF documentation shows a setting for 9 databits (oddly named UART_DATA_BITS_MAX) with which the communication should be possible. Would it be too much to ask for suitable R values? Again, sorry, but my brain works in 1s and 0s whereas power is yellow and is generated by wall plugs ;)

Regards
Jan-Gerd

**Comment by AK-Homberger** [comment]
https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/uart.html

According to the reference the maximum is 8 bit. The problem with 9 bit is that it doubles the required memory for the ring buffer.

But there might be a SoftwareSerial library available thet supports also 9 bit protocol. But I never tried with the ESP32: https://esp32.com/viewtopic.php?t=8835

**Reply by jangerd** [reply]
You are right, I misinterpreted the maximum which they use as a strict upper bound. Thank you for the discussion and the link above!
